In [1]:
import pandas as pd

In [2]:
rawDF=pd.read_csv("social_media_survey.csv")

rawDF

Timestamp  \
0   1/23/2024 12:09:55   
1   1/23/2024 12:22:04   
2   1/23/2024 12:22:19   
3   1/23/2024 12:23:51   
4   1/23/2024 12:24:45   
5   1/23/2024 12:25:41   
6   1/23/2024 12:31:12   
7   1/23/2024 14:23:38   
8   1/23/2024 15:58:17   
9   1/23/2024 16:07:04   
10  1/24/2024 12:49:05   

   Which of the following social media platforms do you use? (Pick all that apply)  \
0                    Instagram, Twitter / X, Snapchat                                
1               Instagram, Facebook, TikTok, Snapchat                                
2   Instagram, Twitter / X, Facebook, TikTok, Snap...                                
3                         Instagram, TikTok, Snapchat                                
4                         Instagram, TikTok, Snapchat                                
5   Instagram, Twitter / X, Facebook, TikTok, Snap...                                
6   Instagram, Twitter / X, Facebook, TikTok, Snap...                                
7                                    Facebook, TikTok                                
8                                    Facebook, TikTok                                
9                                    Facebook, TikTok                                
10              Instagram, Facebook, TikTok, Snapchat                                

   Which one of the following social media platforms do you use the most?  \
0                                           Instagram                       
1                                            Snapchat                       
2                                            Snapchat                       
3                                              TikTok                       
4                                           Instagram                       
5                                           Instagram                       
6                                              TikTok                       
7                                              TikTok                       
8                                            Facebook                       
9                                            Facebook                       
10                                          Instagram                       

   When do you most often use social media?  \
0                                     Night   
1                                     Night   
2                                     Night   
3                                     Night   
4                                   Evening   
5                                   Evening   
6                                     Night   
7                                 Afternoon   
8                                     Night   
9                                   Evening   
10                                    Night   

   Why do you mostly use social media at the time of day chosen above?  \
0   Availability (No other responsibilities at tha...                    
1   Availability (No other responsibilities at tha...                    
2   Availability (No other responsibilities at tha...                    
3   Availability (No other responsibilities at tha...                    
4   Availability (No other responsibilities at tha...                    
5   Availability (No other responsibilities at tha...                    
6   Availability (No other responsibilities at tha...                    
7   Complimenting with another action during that ...                    
8   Availability (No other responsibilities at tha...                    
9   Availability (No other responsibilities at tha...                    
10  Availability (No other responsibilities at tha...                    

   Where do you most often use social media?  \
0                                       Home   
1                                       Home   
2                                       Home   
3                                       Home   
4                                       Home   


In [9]:
most_common_age_group = rawDF['What is your age group?'].value_counts().idxmax()
most_common_age_group


'21-25'

In [7]:
age_group_percentage = rawDF['What is your age group?'].value_counts(normalize=True) * 100
age_group_percentage

What is your age group?
21-25    36.363636
16-20    18.181818
50+      18.181818
46-50     9.090909
26-30     9.090909
31-35     9.090909
Name: proportion, dtype: float64

ChatGPT used for last two entries and code used throughout the project:
find the most common age group out of the survey

1 / 2

ChatGPT
To find the most common age group in the survey, you can use the value_counts() method to count the occurrences of each unique value in the "What is your age group?" column. Here's how you can do it:

python
Copy code
most_common_age_group = rawDF['What is your age group?'].value_counts().idxmax()
print(most_common_age_group)
This code will output the most common age group in the survey. The value_counts() method counts the occurrences of each unique value, and idxmax() returns the index (in this case, the age group) with the maximum count.

Note: If the age group is represented as a range (e.g., '21-25'), the output will be the age group with the highest count, regardless of specific values within the range.

User
adjust the code for jupyter notebooks
ChatGPT
Certainly! In a Jupyter Notebook, you can run the code in a cell. Here's the adjusted code:

python
Copy code
# Assuming rawDF is your DataFrame
# Calculate the most common age group
most_common_age_group = rawDF['What is your age group?'].value_counts().idxmax()
most_common_age_group
In a Jupyter Notebook, the last expression in a cell will be automatically displayed. So, running this cell will display the most common age group in the survey.

User
how do I find the percentage of entries for each age group
ChatGPT
You can find the percentage of entries for each age group by using the value_counts() method to count the occurrences of each unique age group and then dividing each count by the total number of entries in the DataFrame. Here's the code:

python
Copy code
# Assuming rawDF is your DataFrame
# Calculate the percentage of entries for each age group
age_group_percentage = rawDF['What is your age group?'].value_counts(normalize=True) * 100
age_group_percentage
This code uses the value_counts() method with normalize=True to get the relative frequencies (proportions) of each unique value in the "What is your age group?" column. Multiplying by 100 gives the percentage values. The resulting Series will display the percentage of entries for each age group.

In [10]:
gender_percentage = rawDF['What is your gender?'].value_counts(normalize=True) * 100
gender_percentage

What is your gender?
Male      63.636364
Female    36.363636
Name: proportion, dtype: float64

In [11]:
employment_percentage = rawDF['Are you currently employed?'].value_counts(normalize=True) * 100
employment_percentage

Are you currently employed?
Yes    90.909091
No      9.090909
Name: proportion, dtype: float64

In [14]:
# Assuming rawDF is your DataFrame
# Split the entries in the social media platforms column
platforms_split = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].str.split(', ')

# Use explode to create a new row for each platform
exploded_platforms = platforms_split.explode()

# Calculate the percentage of entries for each social media platform
platform_percentage = exploded_platforms.value_counts(normalize=True) * 100

# Create a DataFrame with platform names and percentages
platform_df = pd.DataFrame({'Platform': platform_percentage.index, 'Percentage': platform_percentage.values})
platform_df

Platform  Percentage
0       TikTok   26.315789
1    Instagram   21.052632
2     Snapchat   21.052632
3     Facebook   21.052632
4  Twitter / X   10.526316

ChatGPT used above and code used throughout:
platform_percentage = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].value_counts(normalize=True) * 100
platform_percentage  // I want this to show the individual percentages of each platform
ChatGPT
If you want to display the individual percentages for each social media platform, you can use the following code:

python
Copy code
# Assuming rawDF is your DataFrame
# Calculate the percentage of entries for each social media platform
platform_percentage = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].value_counts(normalize=True) * 100
platform_percentage
This code will output a Series where each platform is associated with its respective percentage of entries in the "Which of the following social media platforms do you use?" column. Each row in the result represents a social media platform, and its corresponding value is the percentage of entries for that platform.

User
that returns this output, but I want it to say Twitter then its response amount, and so on and so forth Which of the following social media platforms do you use? (Pick all that apply)
Instagram, Twitter / X, Facebook, TikTok, Snapchat    27.272727
Facebook, TikTok                                      27.272727
Instagram, Facebook, TikTok, Snapchat                 18.181818
Instagram, TikTok, Snapchat                           18.181818
Instagram, Twitter / X, Snapchat                       9.090909
Name: proportion, dtype: float64
ChatGPT
If you want to display the platform names along with their corresponding percentages, you can create a DataFrame for better readability. Here's how you can modify the code:

python
Copy code
# Assuming rawDF is your DataFrame
# Calculate the percentage of entries for each social media platform
platform_counts = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].value_counts()
platform_percentage = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].value_counts(normalize=True) * 100

# Create a DataFrame with platform names and percentages
platform_df = pd.DataFrame({'Platform': platform_percentage.index, 'Percentage': platform_percentage.values, 'Count': platform_counts.values})
platform_df = platform_df[['Platform', 'Count', 'Percentage']]

platform_df
This code creates a DataFrame (platform_df) with columns for the platform names, their counts, and their percentages. The DataFrame is then displayed with the information you requested.

User
it still returns multiple social medias 
ChatGPT
If you want to separate the multiple social media entries and show individual percentages for each platform, you can split the entries in the "Which of the following social media platforms do you use? (Pick all that apply)" column and then use explode to create a new row for each platform. Here's the modified code:

python
Copy code
# Assuming rawDF is your DataFrame
# Split the entries in the social media platforms column
platforms_split = rawDF['Which of the following social media platforms do you use? (Pick all that apply)'].str.split(', ')

# Use explode to create a new row for each platform
exploded_platforms = platforms_split.explode()

# Calculate the percentage of entries for each social media platform
platform_percentage = exploded_platforms.value_counts(normalize=True) * 100

# Create a DataFrame with platform names and percentages
platform_df = pd.DataFrame({'Platform': platform_percentage.index, 'Percentage': platform_percentage.values})
platform_df
This code first splits the entries in the "Which of the following social media platforms do you use?" column using , as the separator. Then, it uses the explode function to create a new row for each platform. Finally, it calculates the percentage of entries for each social media platform and creates a DataFrame (platform_df) with the platform names and percentages.

In [15]:
favplatform_percentage = rawDF['Which one of the following social media platforms do you use the most?'].value_counts(normalize=True) * 100
favplatform_percentage

Which one of the following social media platforms do you use the most?
Instagram    36.363636
TikTok       27.272727
Snapchat     18.181818
Facebook     18.181818
Name: proportion, dtype: float64

In [16]:
when_percentage = rawDF['When do you most often use social media?'].value_counts(normalize=True) * 100
when_percentage

When do you most often use social media?
Night        63.636364
Evening      27.272727
Afternoon     9.090909
Name: proportion, dtype: float64

In [17]:
where_percentage = rawDF['Where do you most often use social media?'].value_counts(normalize=True) * 100
where_percentage

Where do you most often use social media?
Home    100.0
Name: proportion, dtype: float64

In [18]:
whyloc_percentage = rawDF['Why do you most often use social media at the location selected above?'].value_counts(normalize=True) * 100
whyloc_percentage

Why do you most often use social media at the location selected above?
More time for social media at location    100.0
Name: proportion, dtype: float64

In [19]:
whytime_percentage = rawDF['Why do you mostly use social media at the time of day chosen above?'].value_counts(normalize=True) * 100
whytime_percentage

Why do you mostly use social media at the time of day chosen above?
Availability (No other responsibilities at that time)                                          90.909091
Complimenting with another action during that time (Eating, Mundane Tasks, Work days, etc.)     9.090909
Name: proportion, dtype: float64

In [22]:
reason_percentage = rawDF['Of the following, which is the biggest reason why you use social media? '].value_counts(normalize=True) * 100
reason_percentage

Of the following, which is the biggest reason why you use social media? 
Keeping up with family & friends    54.545455
Entertainment                       45.454545
Name: proportion, dtype: float64